### DISCUSSION TITLE
How to concatenate two string variables?

### DISCUSSION THREAD

**Discussion by hooklab** [bpftrace/bpftrace#2271] [General]
How to concatenate two string variables?

E.g 
$name="Lili"; 
$hi = "Hello"; 
Expected: $out = $hi . $name;
output: Hello Lili

**Comment by xh4n3** [comment]
@oreak If you intend to print that concatenated string, you should use `printf("%s%s", str1, str2)`.

**Reply by hooklab** [reply]
> @oreak If you intend to print that concatenated string, you should use `printf("%s%s", str1, str2)`.

thanks,  I know how to print ,But I want to print every 500 logs and write to some file, so I need a big string concatenation.

**Comment by viktormalik** [comment]
AFAIK bfptrace doesn't have string concatenation operator/builtin. The problem is that string variables are built on the BPF stack which is 512B, so that limits your use-case here. I'm not sure what you mean by "printing logs" but I think that your only option is to print the strings and possibly redirect bpftrace output to a file (using the `-o` option).

**Reply by hooklab** [reply]
> AFAIK bfptrace doesn't have string concatenation operator/builtin. The problem is that string variables are built on the BPF stack which is 512B, so that limits your use-case here. I'm not sure what you mean by "printing logs" but I think that your only option is to print the strings and possibly redirect bpftrace output to a file (using the `-o` option).

Thank you, I know how to solve it, bpftrace itself has a log buffer.
